coordinator/core/core.go

```diff
@@ -554,7 +538,14 @@ func (c *Core) generateCertificateForSecret(secret manifest.Secret, parentCertif
 
 		template.NotAfter = time.Now().AddDate(0, 0, int(secret.ValidFor))
 	} else if secret.ValidFor != 0 {
-		return manifest.Secret{}, errors.New("ambigious certificate validity duration, both NotAfter and ValidFor are specified")
+		if uint(template.NotAfter.Sub(oldBefore).Hours()/24) != secret.ValidFor {
+			// The duration of NotBefore to NotAfter did not match ValidFor, meaning both were specified in the manifest
+			return manifest.Secret{}, errors.New("ambigious certificate validity duration, both NotAfter and ValidFor are specified")
+		}
+		// reset expiration date for private secrets
+		if !secret.Shared {
+			template.NotAfter = time.Now().AddDate(0, 0, int(secret.ValidFor))
```

Comment:
That way we're handling NotBefore and NotAfter implicitly as ValidFor.
However, if I set `NotAfter` in the manifest I would not expect, this to be a dynamic value.
I'd rather explicitly mark `ValidFor` as an overwrite for `NotBefore`/`NotAfter` or always throw an error.

Reply:
If `ValidFor` is set in the manifest, the expiration date will always be that specific date. I think having an option to do this is important.
The case we see here, where `NotAfter` gets dynamically created using `ValidFor`, wont happen if only `ValidFor` is set.

Reply:
The scenario I'm thinking here is the following:
* I have a manifest with a `ValidFor` value set (e.g. 30 days)
* I want to change my deployment to be only valid in a certain time frame. (e.g. 01. August - 31. August) and set `NotBefore` and `NotAfter` accordingly. However, I forget to delete the `ValidFor` set.
* For some reason a pod restarts on 10. August
* I'd expect this pod to have a certificate valid until 31. August, but it's valid from 10. August - 09. September

This is what I meant with implicit